src/_guides/language/language-tour_migrated.md

```diff
@@ -363,6 +363,51 @@ assert(lineCount == null);
   _condition_ is false. For details, see [Assert](#assert).
 {{site.alert.end}}
 
+
+### Late variables
+
+Dart 2.12 added the `late` modifier, which has two use cases:
+
+* To tell Dart that a non-nullable variable is set before it's used.
+* To lazily initialize a variable.
+
+Often Dart's control flow analysis can detect when a non-nullable variable
+is set to a non-null value before it's used,
+but sometimes analysis fails.
+One common case is top-level variables:
+Dart often can't determine whether they're set,
+so it doesn't try.
+
+If you're sure that a variable is set before it's used,
+but Dart disagrees,
+you can fix the error by marking the variable as `late`:
+
+<?code-excerpt "../null_safety_examples/misc/lib/language_tour/variables.dart (var-late-top-level)" replace="/late/[!$&!]/g"?>
+```dart
+[!late!] String description;
+
+void main() {
+  description = 'Feijoada!';
+  print(description);
+}
+```
+
+Use `late` for lazy initialization when a variable might not be needed,
+and initializing it is costly.
+When you mark a variable as `late` but initialize it at its declaration,
+then the initializer doesn't run until the first time the variable is used.
```

Comment:
A nice benefit of lazy initialization is that late instance fields do have access to `this`, unlike other instance fields. It might be worth spelling that out, since it's not an obvious capability, but is pretty useful.